src/include/sof/sof.h

```diff
@@ -48,6 +48,11 @@ struct sa;
 #define PP_NARG(...) (sizeof((unsigned int[]){0, ##__VA_ARGS__}) \
 	/ sizeof(unsigned int) - 1)
 
+/* compile-time assertion */
+#define STATIC_ASSERT(COND, MESSAGE)\
+	__attribute__((unused))\
+	typedef char assertion_failed_##MESSAGE[(COND) ? 1 : -1]
+
```

Comment:
Merged https://github.com/thesofproject/sof/pull/478 into this PR.
Otherwise @akloniex  would have to merge needlessly to me.

Reply:
@lgirdwood already merged this.